### PR TITLE
Try to reuse existing Release object

### LINKED ISSUE #10: Duplicated releases for multiple webhooks
If a release is deleted in GitHub and a new one is made with the same name, this creates multiple Release objects for essentially the same release.

We should detect if there is an existing object that matches the name and try to reuse it.

### PR DESCRIPTION
**Pull request by cpolcyn**
Fixes #10

### CODE REVIEW

File: statusite/repository/views.py [hunk 1]
```diff
@@ -63,25 +63,26 @@ def github_release_webhook(request):
     except Repository.DoesNotExist:
         return HttpResponse('Not listening for this repository')
 
-    time_created = dateutil.parser.parse(release_event['release']['created_at'])
+    try:
+        release = Release.objects.get(version=release_event['release']['name'])
```
Comment (cdcarter) [inline statusite/repository/views.py]: forgive me if this is dumb but is version name unique? shouldn't this be scoped to the repo, at least?

Comment (cpolcyn) [inline statusite/repository/views.py]: Not dumb, good catch! Yes should be scoped to owner/repo as well and probably raise exception when multiple found (not sure if get() already does this for us)

Comment (cpolcyn) [inline statusite/repository/views.py]: Note that duplicate version values will not exist after this change because they would be reused.

Comment (cdcarter) [inline statusite/repository/views.py]: get() will throw an exception (HTTP 500 unless you deal with it), get_object_or_404 will throw a 404 on multiple values

**Comment by cpolcyn** [comment]
@cdcarter I made the requested changes, and added a data migration to delete duplicate releases (keeps the one with most recent time_added). Please do another review when you have a minute. Thanks!

**Pull request merged**